### PR TITLE
Updated license template

### LINKED ISSUE #656: License template update
Due to changes to cmip6-cmor-tables from https://github.com/PCMDI/cmip6-cmor-tables/pull/360 and https://github.com/WCRP-CMIP/CMIP6_CVs/pull/1066, the way CMOR processes the license template will need to be changed.

The license template will be changed from the one below...
```
    "license":[
        "^CMIP6 model data produced by .* is licensed under a Creative Commons Attribution.*ShareAlike 4.0 International License .https://creativecommons.org/licenses.* *Consult https://pcmdi.llnl.gov/CMIP6/TermsOfUse for terms of use governing CMIP6 output, including citation requirements and proper acknowledgment\\. *Further information about this data, including some limitations, can be found via the further_info_url (recorded as a global attribute in this file).*\\. *The data producers and data providers make no warranty, either express or implied, including, but not limited to, warranties of merchantability and fitness for a particular purpose\\. *All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law\\.$"
    ]
```
to this one..
```
    "license":{
        "license":"CMIP6 model data produced by <Your Institution; see CMIP6_institution_id.json> is licensed under a <Creative Commons; select and insert a license_id; see below> License (<insert the matching license_url; see below>). Consult https://pcmdi.llnl.gov/CMIP6/TermsOfUse for terms of use governing CMIP6 output, including citation requirements and proper acknowledgment. Further information about this data, including some limitations, can be found via the further_info_url (recorded as a global attribute in this file)[ and at <some URL maintained by modeling group>]. The data producers and data providers make no warranty, either express or implied, including, but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law.",
        "license_options":{
            "CC BY 4.0":{
                "license_id":"Creative Commons Attribution 4.0 International",
                "license_url":"https://creativecommons.org/licenses/by/4.0/"
            },
            "CC BY-NC-SA 4.0":{
                "license_id":"Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International",
                "license_url":"https://creativecommons.org/licenses/by-nc-sa/4.0/"
            },
            "CC BY-SA 4.0":{
                "license_id":"Creative Commons Attribution-ShareAlike 4.0 International",
                "license_url":"https://creativecommons.org/licenses/by-sa/4.0/"
            },
            "CC0 1.0":{
                "license_id":"Creative Commons CC0 1.0 Universal Public Domain Dedication",
                "license_url":"https://creativecommons.org/publicdomain/zero/1.0/"
            }
        }
    }
```

### PR DESCRIPTION
**Pull request by mauzey1**
Resolves #656 

This PR includes the changes from [cmip6-cmor-tables](https://github.com/PCMDI/cmip6-cmor-tables/tree/1ef31602d3b22b758def6b6cf13fc0793bfcebe0) (https://github.com/PCMDI/cmip6-cmor-tables/pull/360) that will add the new license templates.  No new changes were needed in CMOR code, but the user input for the tests needed to have their license strings changed.  This PR also fixes an issue in [test_python_CMIP6_CV_externalvariables.py](https://github.com/PCMDI/cmor/compare/updated_license_template?expand=1#diff-0440ee2b101bbc2f70076601f6ce8697320b67b7f1c2264eeba779fe8586092f) where redundant calls to cmor.close were causing errors when getting the names of test files to delete.

### CODE REVIEW

File: Test/CMOR_input_TestTables.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
...
+    "license":                 "CMIP6 model data produced by Lawrence Livermore PCMDI is licensed under a Creative Commons Attribution-ShareAlike 4.0 International License (https://creativecommons.org/licenses/by-sa/4.0/). Consult https://pcmdi.llnl.gov/CMIP6/TermsOfUse for terms of use governing CMIP6 output, including citation requirements and proper acknowledgment. Further information about this data, including some limitations, can be found via the further_info_url (recorded as a global attribute in this file) and at https:///pcmdi.llnl.gov/. The data producers and data providers make no warranty, either express or implied, including, but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law.",
```
Comment (durack1) [inline Test/CMOR_input_TestTables.json:40]: @mauzey1 I wonder whether we should update this to the `CC BY 4.0` (not the `-SA` variant) to be consistent, even though this is a test
https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_license.json#L3

Comment (durack1) [reply Test/CMOR_input_TestTables.json:40]: The compressed version of this license string is "Creative Commons Attribution 4.0 International (CC BY 4.0; https://creativecommons.org/licenses/by/4.0/) License"

https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_source_id.json#L19

Comment (mauzey1) [reply Test/CMOR_input_TestTables.json:40]: Do you want this change applied to all the CMOR user input used by tests?  Or just this file?

Comment (mauzey1) [reply Test/CMOR_input_TestTables.json:40]: @durack1 Shouldn't the string be "Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/)"?

Comment (durack1) [reply Test/CMOR_input_TestTables.json:40]: @mauzey1 the license identifier is "Creative Commons Attribution 4.0 International" and the info in parentheses points to details, so I am happy with the formatting.

Regarding the extra work, if this won't impact any of the tests, then I suggest we only change a small number - if you foresee it causing problems if it is inconsistent, then the extra work is warranted

**Review by matthew-mizielinski — COMMENTED** [review]
Changes look good to me from inspection (caveat: I haven't run the tests). I was surprised that CMOR already had the capability to handle multiple license templates/patterns, but I have verified that this works in v3.6.1

**Review by durack1 — APPROVED** [review]
@mauzey1 if these changes will not cause problems downstream, let's get this merged

**Pull request merged**